diff --git a/gyb.py b/gyb.py
index 45388ea..81c0a9f 100755
--- a/gyb.py
+++ b/gyb.py
@@ -47,6 +47,7 @@ import shlex
 import urlparse
 from itertools import islice, chain
 import math
+import errno
 
 import atom.http_core
 import gdata
@@ -154,11 +155,15 @@ def SetupOptionParser():
   return parser
 
 def getProgPath():
-  if os.path.abspath('/') != -1:
-    divider = '/'
-  else:
-    divider = '\\'
-  return os.path.dirname(os.path.realpath(sys.argv[0]))+divider
+	path = os.environ.get('XDG_CONFIG_HOME', os.path.expanduser('~/.config')) + '/gyb/'
+	try:
+		os.makedirs(path)
+	except OSError as exc:
+		if exc.errno == errno.EEXIST and os.path.isdir(path):
+			pass
+		else:
+			raise
+	return path
 
 def batch(iterable, size):
   sourceiter = iter(iterable)
